### PR TITLE
erasure-code: fix a compile error on libec_jerasure

### PR DESCRIPTION
**Pull request by dongbula**
Fixes: https://tracker.ceph.com/issues/65652

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/erasure-code/jerasure/jerasure_init.cc [hunk 1]
```diff
@@ -18,13 +18,15 @@
 #include "common/debug.h"
 #include "jerasure_init.h"
 
+#ifdef __cplusplus
```
Comment (tchaikov) [inline src/erasure-code/jerasure/jerasure_init.cc:21]: this is not necessary. .cc file is compiled as C++ source file by CMake.

File: src/erasure-code/jerasure/jerasure_init.cc [hunk 2]
```diff
@@ -18,13 +18,15 @@
...
 extern "C" {
+#endif
+
 #include "galois.h"
-}
 
 #define dout_context g_ceph_context
 
-extern "C" int jerasure_init(int count, int *words)
+int jerasure_init(int count, int *words)
```
Comment (tchaikov) [inline src/erasure-code/jerasure/jerasure_init.cc:29]: i don't understand why we need this fix or what this fix addresses. please elaborate in the commit message.

File: src/erasure-code/jerasure/jerasure_init.cc (hunk 1 above)
Comment (dongbula) [reply src/erasure-code/jerasure/jerasure_init.cc:21]: I compile it on ubuntu 22, and find vstart.sh will stop after print "load dlopen(ceph/build/lib/libec_jerasure.so): /ceph/build/lib/libec_jerasure.so: undefined symbol: jerasure_init". test this pr can fix it
is it ok?

Comment (tchaikov) [reply src/erasure-code/jerasure/jerasure_init.cc:21]: > is it ok?

it's okay. but it's not enough. you need to convince the reviewer by reasoning not by observations.

Comment (dongbula) [reply src/erasure-code/jerasure/jerasure_init.cc:21]: I runned nm, "nm lib/libec_jerasure.so | grep -w jerasure_init" print:
"U jerasure_init"

but i can't see it again after update and recompile, not very sure about it, just close it and try repetition latter

**Pull request closed without merging**